crystal_helper.py

```diff
@@ -0,0 +1,229 @@
+            file.writelines('{}\n'.format(atoms.num_sites))
+
+            #atom number + coordinates cart
+            for i in range(atoms.num_sites):
+                atomic_number = atoms.atomic_numbers[i]
+                atom_coord = ' '.join(str(np.around(n,5)) for n in atoms.cart_coords[i])
+                file.writelines('{} {}\n'.format(atomic_number,atom_coord))
+
+            #space group + n symm ops
+            file.writelines('{} {}'.format(SpacegroupAnalyzer(atoms).get_space_group_number(),len(SpacegroupAnalyzer(atoms).get_space_group_operations())))
+
+#     def pmg_structure_to_geom_inp(self):
+#         geom_block=["CRYSTAL","0 0 0"]
+#         space_group=SpacegroupAnalyzer(self.structure, symprec=0.01, angle_tolerance=0.2)
+#         geom_block.append(str(space_group.get_space_group_number()))
+#         std_conventional=space_group.get_conventional_standard_structure()
+#         lat_params=[str(round(x,4)) if i>2 else str(round(x,3)) for i,x in enumerate(list(std_conventional.lattice.parameters))]
+        
+#         if space_group.get_lattice_type()=='cubic':
+#             geom_block.append(lat_params[0])
+#         elif space_group.get_lattice_type()=='hexagonal' or space_group.get_lattice_type()=='tetragonal':
+#             geom_block.append(lat_params[0]+' '+lat_params[2])
+#         elif space_group.get_lattice_type()=='trigonal':
+#             geom_block.append(lat_params[0]+' '+lat_params[3])
+#         elif space_group.get_lattice_type()=='orthorhombic':
+#             geom_block.append(lat_params[0]+' '+lat_params[1]+' '+lat_params[2])
+#         elif space_group.get_lattice_type()=='monoclinic':
+#             geom_block.append(lat_params[0]+' '+lat_params[1]+' '+lat_params[2]+' '+lat_params[4])
+#         elif space_group.get_lattice_type()=='triclinic':
+#             geom_block.append(' '.join(x) for x in lat_params)
+        
+        
+#         geom_block.append(str(len(self.sym_distinct_sites)))
+#         for site in self.sym_distinct_sites:
+#             geom_block.append(str(site.species.elements[0].Z)+" "+str(round(site.a,5))+" "+str(round(site.b,5))+" "+str(round(site.c,5)))
+#         return geom_block
+    def cry_shrink(self,spacing=0.2):
+        short_vector=np.min(np.array(self.structure.lattice.reciprocal_lattice.lengths))
+        return int(np.ceil(short_vector/spacing))
+
+    def write_cry_input(self):
+        self.write_cry_gui(self.name.replace(".d12",".gui"))
+        geom_block= ["EXTERNAL"]
+        func_block = ["OPTGEOM","PRINTFORCES","MAXCYCLE","1","END\n","END\n"] if self.FORCECALC==True else ["END\n"]
+        bs_block = []
+        for element in self.elements:
+            bs_block += [s for s in basis_sets[currently_used_bs[element-1]].split("\r\n")]
+        bs_block += ['99 0','ENDBS\n']
+        dft_block = ["DFT","B3LYP","END\n"]
+        scf_block = ["TOLINTEG","7 7 7 7 14","SHRINK",str(self.shrink)+" "+str(self.shrink)+" "+str(2*self.shrink),"END"]
```

Comment:
SHRINK only needs two values: one for the scf and one for the Fermi energy calculation